src/state/opamSysInteract.ml

```diff
@@ -1093,14 +1105,51 @@ let install_packages_commands_t ?(env=OpamVariable.Map.empty) config sys_package
     [`AsUser (Commands.msys2 config),
      "-Su"::"--noconfirm"::packages], None
   | Netbsd -> [`AsAdmin "pkgin", yes ["-y"] ("install" :: packages)], None
+  | Nix ->
+    let open OpamFilename in
+    (match switch with
+    | None ->
+        log "Nix depext must be passed switch";
+        [], None
+    | Some switch ->
+    let dir = OpamPath.Switch.meta OpamStateConfig.(!r.root_dir) switch in
+    let drvFile = create dir (basename (raw "env.nix")) in
+    let packages = String.concat " "
+      (OpamSysPkg.Set.fold (fun p l -> OpamSysPkg.to_string p :: l)
+      OpamSysPkg.Set.Op.(sys_packages ++ required) [])
+    in
+    let contents =
+{|{ pkgs ? import <nixpkgs> {} }:
+with pkgs;
+stdenv.mkDerivation {
+  name = "opam-nix-env";
+  nativeBuildInputs = with buildPackages; [ |} ^ packages ^ {| ];
+
+  phases = [ "buildPhase" ];
+
+  buildPhase = ''
+vars=("NIX_CC" "NIX_CC_FLAGS" "NIX_CFLAGS_COMPILE" "NIX_CC_WRAPPER_TARGET_HOST_x86_64_unknown_linux_gnu" "NIX_LDFLAGS" "PKG_CONFIG_PATH")
+for var in "''${vars[@]}"; do
+  escaped="$(echo "''${!var}" | sed -e 's/^$/@/' -e 's/ /\\ /g')"
+  echo "$var	=	$escaped	Nix" >> $out
+done
+echo "PATH	+=	$PATH	Nix" >> $out
```

Comment:
I'm not sure to understand when the bash script starts and end but if those lines are indeed a bash script then these variables need to be quoted
```suggestion
  echo "$var	=	$escaped	Nix" >> "$out"
done
echo "PATH	+=	$PATH	Nix" >> "$out"
```

Reply:
`$out` will be a [Nix store path](https://nix.dev/manual/nix/2.23/store/store-path), so shouldn't need quoted, but I don't see any harm in adding them.